### ISSUE TITLE
Improve handling of checkboxes in _Religion view_

### ISSUE THREAD

**Issue by tutebatti** [UniStuttgart-VISUS/damast#34]
I suggest to add two buttons to check or uncheck all religious groups to facilitate filtering. This should be applied for both Simple and Advanced mode.

Also, is it possible to check/uncheck an entire category of religious groups through a second checkbox belonging to the main religion?

**Comment by mfranke93** [comment]
> I suggest to add two buttons to check or uncheck all religious groups to facilitate filtering. This should be applied for both Simple and Advanced mode.

Yes, absolutely. My suggestion would be to replace the invert button in the column header(s) by the three buttons available in the sources and tags view: none, invert, all. The disadvantage would be that the header would be higher then. One advantage I see right away is that the symbols (invert) would be more consistent.

 > Also, is it possible to check/uncheck an entire category of religious groups through a second checkbox belonging to the main religion?

I think we could do that. Maybe something like this, where parent nodes get extra controls:
![image](https://user-images.githubusercontent.com/9373678/149154025-dd230ca9-c9c0-43ff-b3f9-399e64624832.png)

Of course, all of these changes introduce extra complexity when using the visualization.

**Comment by tutebatti** [comment]
For the overall list, I would really see the advantage in consistency with the other views and an easier control of the checkboxes. For the feature regarding the extra checkboxes for all religions: I see that it absolutely affects the impression of the user, probably being overwhelmed. Two possible improvements to your suggestion: (1) I would position the buttons next to the respective checkbox; (2) I would only have one button switching between "select all" and "select none". "Invert" is probably not needed here. (On the other hand, this is inconsistent again with the other groupings of the buttons...)

**Comment by mfranke93** [comment]
Agreed on invert not being needed. However (and this does not really work with my solution either): How would those controls work in advanced mode? You would need one set of controls for each category (column).

**Comment by tutebatti** [comment]
Good question! The columns would need to be wider, leaving space for the button next to every checkbox somewhat like this:
![grafik](https://user-images.githubusercontent.com/7239968/149157705-5144cba5-3610-449d-a410-56d7f0f7e03c.png)

**Comment by mfranke93** [comment]
Yeah. Of course, that would halve the number of columns that can be used while still having somewhat readable areas in the indented tree. Right now, I would say that about 10 columns fit quite well. After, that would only be five.

**Comment by tutebatti** [comment]
I'm not saying this to push for implementing the extra buttons/checkboxes, but I seriously doubt anyone would use more than 5 columns... :smile:

**Comment by mfranke93** [comment]
Sure. In any case, one could maximize the view for that.

However, that will be a bit of work. Those checkboxes and the entire handling around them is a bit older and the code is a bit messy. A good time to clean that up, but I don't want to make promises as to when I get around to it.

**Comment by tutebatti** [comment]
Again, I suggest to focus on the buttons for the overall list. In general, the whole thing works as is, no problem.

**Comment by tutebatti** [comment]
Marvelous implementation - thank you!
